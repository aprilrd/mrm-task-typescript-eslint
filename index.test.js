jest.mock("fs", () => require("memfs"));
jest.mock("mrm-core/src/util/log", () => ({
  added: jest.fn()
}));
const mockInstall = jest.fn();
jest.mock("mrm-core/src/npm", () => ({
  install: mockInstall
}));

const { getConfigGetter } = require("mrm");
const vol = require("memfs").vol;
const task = require("./index");

const stringify = o => JSON.stringify(o, null, "  ");

afterEach(() => vol.reset());

const scaffold = ({ caseName, fsStatus = {}, eslintConfig }) => {
  describe(`for a ${caseName}`, () => {
    beforeAll(() => {
      vol.fromJSON(fsStatus);

      task(getConfigGetter({ eslintConfig }));
    });

    it(`should add TypeScript ESLint to ${caseName}`, () => {
      expect(vol.toJSON()).toMatchSnapshot();
    });

    it("should install TypeScript ESLint plugins", () => {
      expect(mockInstall).toHaveBeenCalledWith([
        "@typescript-eslint/eslint-plugin",
        "@typescript-eslint/parser"
      ]);
    });
  });
};

[
  {
    caseName: "empty .eslintrc.yaml"
  },
  {
    caseName: "preexisting .eslintrc.yaml",
    fsStatus: {
      "./.eslintrc.yaml": stringify("")
    },
    eslintConfig: "./.eslintrc.yaml"
  },
  {
    caseName: "preexisting .eslintrc.json",
    fsStatus: {
      "./.eslintrc.json": stringify("")
    },
    eslintConfig: "./.eslintrc.json"
  },
  {
    caseName: "preexisting .eslintrc",
    fsStatus: { "./eslintrc": stringify({}) },
    eslintConfig: "./.eslintrc"
  },
  {
    caseName: "preexisting .eslintrc 2",
    fsStatus: {
      "./.eslintrc": stringify({
        extends: ["airbnb", "prettier"],
        parser: "babel-eslint",
        plugins: ["react", "prettier"],
        parserOptions: {
          ecmaFeatures: {
            experimentalObjectRestSpread: 1
          },
          sourceType: "script"
        }
      })
    },
    eslintConfig: "./.eslintrc"
  }
].forEach(scaffold);
