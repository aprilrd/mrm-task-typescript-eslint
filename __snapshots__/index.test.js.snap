// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`for a empty .eslintrc.yaml should add TypeScript ESLint to empty .eslintrc.yaml 1`] = `
Object {
  "/Users/youngmink/projects/personal/mrm-task-typescript-eslint/.eslintrc.yaml": "parser: '@typescript-eslint/parser'
settings:
  import/parsers:
    '@typescript-eslint/parser':
      - .js
      - .jsx
      - .ts
      - .tsx
plugins:
  - '@typescript-eslint'
extends:
  - 'plugin:@typescript-eslint/recommended'
  - 'plugin:import/typescript'
",
}
`;

exports[`for a preexisting .eslintrc 2 should add TypeScript ESLint to preexisting .eslintrc 2 1`] = `
Object {
  "/Users/youngmink/projects/personal/mrm-task-typescript-eslint/.eslintrc": "{
  \\"extends\\": [
    \\"plugin:@typescript-eslint/recommended\\",
    \\"plugin:import/typescript\\",
    \\"airbnb\\",
    \\"prettier\\"
  ],
  \\"parser\\": \\"@typescript-eslint/parser\\",
  \\"plugins\\": [
    \\"react\\",
    \\"prettier\\",
    \\"@typescript-eslint\\"
  ],
  \\"parserOptions\\": {
    \\"ecmaFeatures\\": {
      \\"experimentalObjectRestSpread\\": 1
    },
    \\"sourceType\\": \\"script\\"
  },
  \\"settings\\": {
    \\"import/parsers\\": {
      \\"@typescript-eslint/parser\\": [
        \\".js\\",
        \\".jsx\\",
        \\".ts\\",
        \\".tsx\\"
      ]
    }
  }
}",
}
`;

exports[`for a preexisting .eslintrc should add TypeScript ESLint to preexisting .eslintrc 1`] = `
Object {
  "/Users/youngmink/projects/personal/mrm-task-typescript-eslint/.eslintrc": "{
  \\"parser\\": \\"@typescript-eslint/parser\\",
  \\"settings\\": {
    \\"import/parsers\\": {
      \\"@typescript-eslint/parser\\": [
        \\".js\\",
        \\".jsx\\",
        \\".ts\\",
        \\".tsx\\"
      ]
    }
  },
  \\"plugins\\": [
    \\"@typescript-eslint\\"
  ],
  \\"extends\\": [
    \\"plugin:@typescript-eslint/recommended\\",
    \\"plugin:import/typescript\\"
  ]
}",
  "/Users/youngmink/projects/personal/mrm-task-typescript-eslint/eslintrc": "{}",
}
`;

exports[`for a preexisting .eslintrc.json should add TypeScript ESLint to preexisting .eslintrc.json 1`] = `
Object {
  "/Users/youngmink/projects/personal/mrm-task-typescript-eslint/.eslintrc.json": "{
  \\"parser\\": \\"@typescript-eslint/parser\\",
  \\"settings\\": {
    \\"import/parsers\\": {
      \\"@typescript-eslint/parser\\": [
        \\".js\\",
        \\".jsx\\",
        \\".ts\\",
        \\".tsx\\"
      ]
    }
  },
  \\"plugins\\": [
    \\"@typescript-eslint\\"
  ],
  \\"extends\\": [
    \\"plugin:@typescript-eslint/recommended\\",
    \\"plugin:import/typescript\\"
  ]
}",
}
`;

exports[`for a preexisting .eslintrc.yaml should add TypeScript ESLint to preexisting .eslintrc.yaml 1`] = `
Object {
  "/Users/youngmink/projects/personal/mrm-task-typescript-eslint/.eslintrc.yaml": "parser: '@typescript-eslint/parser'
settings:
  import/parsers:
    '@typescript-eslint/parser':
      - .js
      - .jsx
      - .ts
      - .tsx
plugins:
  - '@typescript-eslint'
extends:
  - 'plugin:@typescript-eslint/recommended'
  - 'plugin:import/typescript'",
}
`;
